homeassistant/components/eq3btsmart/number.py

```diff
@@ -0,0 +1,224 @@
+"""Platform for eq3 number entities."""
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+from eq3btsmart import Thermostat
+from eq3btsmart.const import (
+    EQ3BT_MAX_OFFSET,
+    EQ3BT_MAX_TEMP,
+    EQ3BT_MIN_OFFSET,
+    EQ3BT_MIN_TEMP,
+)
+from eq3btsmart.models import Presets
+from eq3btsmart.thermostat_config import ThermostatConfig
+
+from homeassistant.components.number import (
+    NumberDeviceClass,
+    NumberEntity,
+    NumberEntityDescription,
+    NumberMode,
+    RestoreNumber,
+)
+from homeassistant.const import EntityCategory, UnitOfTemperature, UnitOfTime
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import Eq3ConfigEntry
+from .const import (
+    ENTITY_KEY_AWAY_HOURS,
+    ENTITY_KEY_AWAY_TEMPERATURE,
+    ENTITY_KEY_COMFORT,
+    ENTITY_KEY_ECO,
+    ENTITY_KEY_OFFSET,
+    ENTITY_KEY_WINDOW_OPEN_TEMPERATURE,
+    ENTITY_KEY_WINDOW_OPEN_TIMEOUT,
+    EQ3BT_STEP,
+)
+from .entity import Eq3Entity
+
+
+@dataclass(frozen=True, kw_only=True)
+class Eq3NumberEntityDescription(NumberEntityDescription):
+    """Entity description for eq3 number entities."""
+
+    value_func: Callable[[Presets], float]
+    value_set_func: Callable[
+        [Thermostat],
+        Callable[[float], Awaitable[None]],
+    ]
+    device_class: NumberDeviceClass | None = NumberDeviceClass.TEMPERATURE
+    native_unit_of_measurement: str = UnitOfTemperature.CELSIUS
+    native_min_value: float = EQ3BT_MIN_TEMP
+    native_max_value: float = EQ3BT_MAX_TEMP
+    native_step: float = EQ3BT_STEP
+    mode: NumberMode = NumberMode.BOX
+    entity_category: EntityCategory | None = EntityCategory.CONFIG
```

Comment:
I am not sure if others share this opinion, but I personally think that defaults for entity descriptions should be the default for everything, and not just a default for 3/5 definitions. The charm I like about entity descriptions is that the representation is concentrated in a single place and that is more spread out like this.

But the things like `mode` or `entity_category` are perfect for this